shared_model/interfaces/common_objects/amount.hpp

```diff
@@ -1,84 +1,72 @@
 /**
- * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
- * http://soramitsu.co.jp
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *        http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
  */
 
 #ifndef IROHA_SHARED_MODEL_AMOUNT_HPP
 #define IROHA_SHARED_MODEL_AMOUNT_HPP
 
-#include <boost/multiprecision/cpp_dec_float.hpp>
-#include <boost/multiprecision/cpp_int.hpp>
-#include <vector>
-
 #include "interfaces/base/model_primitive.hpp"
+
+#include <boost/multiprecision/cpp_int.hpp>
 #include "interfaces/common_objects/types.hpp"
-#include "utils/string_builder.hpp"
+#include "utils/lazy_initializer.hpp"
 
 namespace shared_model {
   namespace interface {
 
     /**
      * Representation of fixed point number
      */
-    class Amount : public ModelPrimitive<Amount> {
+    class Amount final : public ModelPrimitive<Amount> {
      public:
+      explicit Amount(const std::string &amount);
+      explicit Amount(std::string &&amount);
+
+      Amount(const Amount &o);
```

Comment:
If you add move and copy constructors please add assignment operators as well

Reply:
With const member it's impossible